src/main/java/com/lgtm/easymoney/exceptions/handlers/ControllerExceptionHandler.java

```diff
@@ -99,4 +100,14 @@ public ResponseEntity<ErrorRsp> handle(InvalidUpdateException ex) {
     return ResponseEntity.status(HttpStatus.BAD_REQUEST)
             .body(new ErrorRsp(errorFields, errorMessage));
   }
+
+  /** This handles dataabse failure. e.g. saving transaction failed */
+  @ExceptionHandler(DatabaseFailureException.class)
+  public ResponseEntity<ErrorRsp> handle(DatabaseFailureException ex) {
+    List<String> errorFields = new ArrayList<>();
+    errorFields.add(ex.getResourceName());
+    String errorMessage = ex.getMessage();
+    return ResponseEntity.status(HttpStatus.BAD_REQUEST)
```

Comment:
I think https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422 is a better status code lol